internal/provider/resource_tfe_test_variable_test.go

```diff
@@ -50,6 +53,40 @@ func TestAccTFETestVariable_basic(t *testing.T) {
 	})
 }
 
+func TestAccTFETestVariable_valueWriteOnly(t *testing.T) {
+	variable := &tfe.Variable{}
+	rInt := rand.New(rand.NewSource(time.Now().UnixNano())).Int()
+
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+			testAccGithubPreCheck(t)
+		},
+		TerraformVersionChecks: []tfversion.TerraformVersionCheck{
+			tfversion.SkipBelow(version.Must(version.NewVersion("1.11.0"))),
+		},
+		ProtoV5ProviderFactories: testAccMuxedProviders,
+		CheckDestroy:             testAccCheckTFETestVariableDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config:      testAccTFETestVariable_valueAndValueWO(rInt),
+				ExpectError: regexp.MustCompile(`Attribute "value" cannot be specified when "value_wo" is specified`),
+			},
+			{
+				Config: testAccTFETestVariable_valueWriteOnly(rInt),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckTFETestVariableExists(
+						"tfe_test_variable.foobar", variable),
+					resource.TestCheckNoResourceAttr(
+						"tfe_test_variable.foobar", "value_wo"),
+					resource.TestCheckResourceAttr(
+						"tfe_test_variable.foobar", "sensitive", "false"),
+				),
+			},
+		},
```

Comment:
There's an opportunity here to use a ValueComparer to assert that the resource is being replaced under certain conditions.

[Here's an example](https://github.com/hashicorp/terraform-provider-tfe/pull/1641/files#diff-3cc46359081aee5c828d76dcb1d6a834a9cbf1a5ff32af6e344e82fce6bdc51aR140-R225) from my PR.